// @freakycoder/react-native-helpers/build/dist/helpers/device/DeviceInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PlatformVersion = exports.WindowFontScale = exports.WindowScale = exports.WindowHeight = exports.WindowWidth = exports.ScreenFontScale = exports.ScreenScale = exports.ScreenHeight = exports.ScreenWidth = exports.isAndroid = exports.isIOS = void 0;
const react_native_1 = require("react-native");
// ? Screen Constants
const Screen = react_native_1.Dimensions.get("screen");
const ScreenWidth = Screen.width;
exports.ScreenWidth = ScreenWidth;
const ScreenHeight = Screen.height;
exports.ScreenHeight = ScreenHeight;
const ScreenScale = Screen.scale;
exports.ScreenScale = ScreenScale;
const ScreenFontScale = Screen.fontScale;
exports.ScreenFontScale = ScreenFontScale;
// ? Window Constants
const Window = react_native_1.Dimensions.get("window");
const WindowWidth = Window.width;
exports.WindowWidth = WindowWidth;
const WindowHeight = Window.height;
exports.WindowHeight = WindowHeight;
const WindowFontScale = Window.fontScale;
exports.WindowFontScale = WindowFontScale;
const WindowScale = Window.scale;
exports.WindowScale = WindowScale;
const isIOS = react_native_1.Platform.OS === "ios";
exports.isIOS = isIOS;
const isAndroid = react_native_1.Platform.OS === "android";
exports.isAndroid = isAndroid;
const PlatformVersion = react_native_1.Platform.Version;
exports.PlatformVersion = PlatformVersion;
